Sources/ConsoleKit/Logging/ConsoleLogger.swift

```diff
@@ -0,0 +1,222 @@
+    ///   - metadata: Extra metadata to log with the message. This defaults to an empty dictionary.
+    ///   - settings: The output configuration this logger should use. This defaults to the default console logger settings.
+    ///   - style: The style configuration this logger should use. This defaults to the default console logger style.
+    ///
+    /// - Note: It would make more sense for the output and style configurations to default to those provided when
+    ///   bootstrapping the logging system (if any). Unfortunately, it was not possible to do this without either
+    ///   breaking existing API or taking the performance hit of a global lock on every logger creation.
+    public init(
+        label: String,
+        console: Console,
+        level: Logger.Level = .debug,
+        metadata: Logger.Metadata = [:],
+        settings: Logger.OutputConfiguration = .default,
+        style: Logger.StyleConfiguration = .default
+    ) {
+        self.label = label
+        self.metadata = metadata
+        self.logLevel = level
+        self.console = console
+        self.outputConfiguration = settings
+        self.styleConfiguration = style
+    }
+
+    /// See `LogHandler[metadataKey:]`.
+    ///
+    /// This just acts as a getter/setter for the `.metadata` property.
+    public subscript(metadataKey key: String) -> Logger.Metadata.Value? {
+        get { return self.metadata[key] }
+        set { self.metadata[key] = newValue }
+    }
+
+    /// See `LogHandler.log(level:message:metadata:source:file:function:line:)`.
+    ///
+    /// - Important: Due to https://github.com/apple/swift-log/issues/145, we currently ignore the `source` parameter,
+    ///   even if it has been manually specified with a non-default value. This will be changed in the future.
+    public func log(
+        level: Logger.Level,
+        message: Logger.Message,
+        metadata: Logger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        /// - TODO: Preallocate a medium-sized buffer that can be reused for building log messages instead of glomming ConsoleTexts together.
+        var text: ConsoleText = ""
+        
+        // Note: The use of `" ".consoleText(self.styleConfiguration.baseStyle)` and similar constructs seems pointless at
+        //       first glance, but its necessity becomes immediately obvious after, for example, giving each of the styles
+        //       a different (and non-default) background color.
+        // TODO: Don't torment the construction of the consoleText pieces so much. This can be done ENORMOUSLY more efficiently.
+
+        // Render the timestamp if currently visible.
+        if self.outputConfiguration.timestampDisplayLogLevel.map({ $0 >= self.logLevel }) ?? false {
+            text += "[".consoleText(self.styleConfiguration.baseStyle) +
```

Comment:
I wonder if the string interpolating API would make this easier to read? For example:
```swift
"[ \("INFO", style: .info) ]"
```